core/t_set.go

```diff
@@ -0,0 +1,290 @@
+
+	var count int = 0
+	if obj != nil {
+		if obj.TypeEncoding != ObjTypeSet|ObjEncodingHT {
+			return Encode(errors.New("WRONGTYPE Operation against a key holding the wrong kind of value"), false)
+		}
+		// create a deep copy of the set object
+		srcSet = make(map[string]bool)
+		for k, flag := range obj.Value.(map[string]bool) {
+			if flag {
+				srcSet[k] = flag
+				count++
+			}
+		}
+	}
+
+	for i := 1; i < len(args); i++ {
+		// Get the set object from the store.
+		obj := Get(args[i])
+
+		if obj == nil {
+			continue
+		}
+
+		// If the object exists, check if it is a set object.
+		if obj.TypeEncoding != ObjTypeSet|ObjEncodingHT {
+			return Encode(errors.New("WRONGTYPE Operation against a key holding the wrong kind of value"), false)
+		}
+
+		// only if the count is greater than 0, we need to check the other sets
+		if count > 0 {
+			// Get the set object.
+			set := obj.Value.(map[string]bool)
+			for k, flag := range set {
+				if flag && srcSet[k] {
+					srcSet[k] = false
+					count--
+				}
+			}
+		}
+	}
+
+	if count == 0 {
+		return Encode([]string{}, false)
+	}
+
+	// Get the members of the set.
+	var members []string
+	for k, flag := range srcSet {
+		if flag {
+			members = append(members, k)
+		}
+	}
+
+	return Encode(members, false)
+}
+
+func evalSINTER(args []string) []byte {
+	if len(args) < 2 {
+		return Encode(errors.New("ERR wrong number of arguments for 'SINTER' command"), false)
+	}
+
+	sets := make([]map[string]bool, len(args))
+
+	var empty int = 0
+
+	for i := 0; i < len(args); i++ {
+		// Get the set object from the store.
+		obj := Get(args[i])
+
+		if obj == nil {
+			empty++
+			continue
+		}
+
+		// If the object exists, check if it is a set object.
+		if obj.TypeEncoding != ObjTypeSet|ObjEncodingHT {
+			return Encode(errors.New("WRONGTYPE Operation against a key holding the wrong kind of value"), false)
+		}
+
+		// Get the set object.
+		set := obj.Value.(map[string]bool)
+		sets[i] = set
+	}
+
+	if empty > 0 {
+		return Encode([]string{}, false)
+	}
+
+	// sort the sets by the number of elements in the set
+	// we will iterate over the smallest set
+	// and check if the element is present in all the other sets
+	sort.Slice(sets, func(i, j int) bool {
+		return len(sets[i]) < len(sets[j])
+	})
+
+	count := 0
+	resultSet := make(map[string]bool)
+	var members []string
+
+	// init the result set with the first set
+	// store the number of elements in the first set in count
+	// we will decrement the count if we do not find the elements in the other sets
+	for k, flag := range sets[0] {
+		if !flag {
+			continue
+		}
+		count++
+		resultSet[k] = true
+	}
+
+	for i := 1; i < len(sets); i++ {
+		if count == 0 {
+			break
+		}
+		for k, flag := range resultSet {
+			if !flag {
+				continue
+			}
+			if !sets[i][k] {
+				delete(resultSet, k)
+				count--
+			}
+		}
+	}
+	for k, flag := range resultSet {
+		if flag {
+			members = append(members, k)
+		}
+	}
+	return Encode(members, false)
```

Comment:
```suggestion
	members := make([]string, 0, len(resultSet))
	for k := range resultSet {
		members = append(members, k)
	}

	return Encode(members, false)
```
Improve readability.